File: app/src/main/java/com/example/android/miwok/WordAdapter.java
package com.example.android.miwok;

import android.app.Activity;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by unknown on 2/4/17
 */

class WordAdapter extends ArrayAdapter<Word>{

    private int mColorResourceId ;

    WordAdapter(Activity context, ArrayList<Word> words, int colorResourceId){
        super(context, 0, words);
        mColorResourceId = colorResourceId ;
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        View listItemView = convertView;
        if(listItemView == null) {
            listItemView = LayoutInflater.from(getContext()).inflate(
                    R.layout.list_item, parent, false);
        }

        Word currentWord = getItem(position);

        // Find the ImageView in the list_item.xml layout with the ID list_item_icon
        ImageView iconView = listItemView.findViewById(R.id.icon);
        // Get the image resource ID from the current AndroidFlavor object and
        // set the image to iconView
        if(currentWord.hasImage()){
            iconView.setImageResource(currentWord.getImageResourceId());
            iconView.setVisibility(View.VISIBLE);
        }
        else {
            iconView.setVisibility(View.GONE);
        }

        // Find the TextView in the list_item.xml layout with the ID miwok_text_view
        TextView miwokTextView = listItemView.findViewById(R.id.miwok_text_view);
        // Get the miwok name from the current Word object and
        // set this text on the name TextView
        miwokTextView.setText(currentWord.getMiwokTranslation());

        // Find the TextView in the list_item.xml layout with the ID version_number
        TextView englishTextView = listItemView.findViewById(R.id.default_text_view);
        // Get the version number from the current AndroidFlavor object and
        // set this text on the number TextView
        englishTextView.setText(currentWord.getDefaultTranslation());

        // Set the theme color for the list item
        RelativeLayout textContainer = listItemView.findViewById(R.id.text_and_image_container);
        // Find the color that the resource ID maps to
        int color = ContextCompat.getColor(getContext(), mColorResourceId);
        // Set the background color of the text container View
        textContainer.setBackgroundColor(color);

        return listItemView;
    }
}

File: app/src/main/java/com/example/android/miwok/WordAdapter.kt
package com.example.android.miwok

import android.app.Activity
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.list_item.view.*
import java.util.*

/**
 * Created by unknown on 2/4/17
 */

internal class WordAdapter(context: Activity, words: ArrayList<Word>, private var colorResourceId: Int) : ArrayAdapter<Word>(context, 0, words) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        var listItemView = convertView
        if (listItemView == null) {
            listItemView = LayoutInflater.from(context).inflate(
                    R.layout.list_item, parent, false)
        }

        val currentWord = getItem(position)

        // Find the ImageView in the list_item.xml layout with the ID list_item_icon
        val iconView = listItemView!!.listViewImageViewAvatar
        // Get the image resource ID from the current AndroidFlavor object and
        // set the image to iconView
        if (currentWord!!.hasImage()) {
            iconView.setImageResource(currentWord.imageResourceId)
            iconView.visibility = View.VISIBLE
        } else {
            iconView.visibility = View.GONE
        }

        // Find the TextView in the list_item.xml layout with the ID miwok_text_view
        val miwokTextView = listItemView.notMiwokTranslationTextView
        // Get the miwok name from the current Word object and
        // set this text on the name TextView
        miwokTextView.text = currentWord.miwokTranslation

        // Find the TextView in the list_item.xml layout with the ID version_number
        val englishTextView = listItemView.miwokTranslationTextView
        // Get the version number from the current AndroidFlavor object and
        // set this text on the number TextView
        englishTextView.text = currentWord.defaultTranslation

        // Set the theme color for the list item
        val textContainer = listItemView.listViewLayoutContainer
        // Find the color that the resource ID maps to
        val color = ContextCompat.getColor(context, colorResourceId)
        // Set the background color of the text container View
        textContainer.setBackgroundColor(color)

        return listItemView
    }
}